src/dhcp.c

```diff
@@ -301,11 +301,19 @@ void dhcp_update(dhcp_t *dhcp,
     dhcp_message_t *msg = NULL;
     const unsigned char magic_cookie[] = {0x63, 0x82, 0x53, 0x63};
 
+    joy_log_debug("dhcp[%p],header[%p],data[%p],len[%d],report[%d]",
+            dhcp,header,data,data_len,report_dhcp);
+
     /* Check run flag. Bail if 0 */
     if (!report_dhcp) {
         return;
     }
 
+    /* sanity check */
+    if (dhcp == NULL) {
+        return;
```

Comment:
Shouldn't we log something here?

Reply:
We log an error if the malloc fails on init. Do we really need to log an error continually down the line? It would just be noisy and its the same error. Really the only way it is NULL at this point is if the malloc failed in init.

Reply:
I agree. No log needed here.